EngineLayer/CompactPeptideWithModifiedMass.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Proteomics.Fragmentation;
+using Proteomics.ProteolyticDigestion;
+
+namespace EngineLayer
+{
+    //[Serializable]
+    //public class CompactPeptideWithModifiedMass : CompactPeptideBase
+    //{
+
+    //    public CompactPeptideWithModifiedMass(CompactPeptideBase cp, double monoisotopicMassIncludingFixedMods)
+    //    {
+    //        CTerminalMasses = cp.CTerminalMasses;
+    //        NTerminalMasses = cp.NTerminalMasses;
+    //        MonoisotopicMassIncludingFixedMods = cp.MonoisotopicMassIncludingFixedMods;
+    //        ModifiedMass = monoisotopicMassIncludingFixedMods;
+    //    }
+
+    //    public double ModifiedMass { get; set; }
+
+    //    public void SwapMonoisotopicMassWithModifiedMass()
+    //    {
+    //        double tempDouble = MonoisotopicMassIncludingFixedMods;
+    //        MonoisotopicMassIncludingFixedMods = ModifiedMass;
+    //        ModifiedMass = tempDouble;
+    //    }
+
+    //    public void CropTerminalMasses(FragmentationTerminus terminusType)
+    //    {
+    //        List<double> tempList = new List<double>();
+    //        double[] masses = terminusType == FragmentationTerminus.N ? NTerminalMasses : CTerminalMasses;
+    //        for (int i = 0; i < masses.Length; i++)
+    //        {
+    //            if (masses[i] < MonoisotopicMassIncludingFixedMods)
+    //            {
+    //                tempList.Add(masses[i]);
+    //            }
+    //            else if (terminusType == FragmentationTerminus.N)
+    //            {
+    //                NTerminalMasses = tempList.ToArray();
+    //                break;
+    //            }
+    //            else
+    //            {
+    //                CTerminalMasses = tempList.ToArray();
+    //                break;
+    //            }
+    //        }
+    //    }
+    //}
+
+    [Serializable]
+    public class PeptideWithSetModifications_ModifiedMass
+    {
+        public double[] CterminalMasses { get; private set; }
+        public double[] NterminalMasses { get; private set; }
+        public double PeptideWithSetModifications_MonoisotopicMass { get; private set; }
+        public double ModifiedMass { get; private set; }
+
+
+        public PeptideWithSetModifications_ModifiedMass(PeptideWithSetModifications p, double newMonoMass, MassSpectrometry.DissociationType d)
```

Comment:
This is unreferenced. Do we need it?

Reply:
I don't think this class is needed anymore. It previously existed for converting compact peptides into pwsms.